simd/src/Kokkos_SIMD_AVX512.hpp

```diff
@@ -980,6 +980,196 @@ class simd<double, simd_abi::avx512_fixed_size<8>> {
                            static_cast<__m512d>(b)));
 }
 
+template <>
+class simd<float, simd_abi::avx512_fixed_size<8>> {
+  __m256 m_value;
+
+ public:
+  using value_type = float;
+  using abi_type   = simd_abi::avx512_fixed_size<8>;
+  using mask_type  = simd_mask<value_type, abi_type>;
+  using reference  = value_type&;
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION simd()            = default;
```

Comment:
```suggestion
  simd() = default;
```

Reply:
(all of these can be cleaned up later)

Reply:
Looking at this again, I am slightly confused on these suggested changes.

These defaulted constructors and operators are also available for simd with `double` type. 

https://github.com/kokkos/kokkos/blob/de4380e9b041d646dfadc6c3ce671d898afac9f7/simd/src/Kokkos_SIMD_AVX512.hpp#L771-L787

Was there a reason why these should be removed for `float` type?